include/SFML/Graphics/Font.hpp

```diff
@@ -124,7 +117,7 @@ class SFML_GRAPHICS_API Font
     /// \see loadFromMemory, loadFromStream
     ///
     ////////////////////////////////////////////////////////////
-    [[nodiscard]] bool loadFromFile(const std::filesystem::path& filename);
+    static std::optional<Font> loadFromFile(const std::filesystem::path& filename);
```

Comment:
`std::optional` itself is not `[[nodiscard]]` to my knowledge -- so should these methods maybe add such an attribute?

After all, C++ even allows static methods to be called on instances (possibly with warnings, depends on QoI).
We could thus avoid mistakes of the form:
```cpp
// reload the font
myExistingFont.loadFromFile("bla");
```

Reply:
+1, good point for the `.loadFromFile` issue.